Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/Contract.cs

```diff
@@ -43,12 +43,20 @@ public static Contract Prepare(Guid customerId, int customerAge, int customerHei
             StandardDuration);
     }
 
-    public void Sign(DateTimeOffset signedAt, DateTimeOffset today)
+    internal BindingContract Sign(DateTimeOffset signedAt, DateTimeOffset dateNow)
     {
+        if (IsSigned)
+        {
+            throw new ContractAlreadySignedException();
```

Comment:
@meaboutsoftware  we need such validation to avoid situation that we have two binding contracts.

I'm thinking about transforming it to business rule

Reply:
Done